### PR TITLE
Issue CORE-335 | Controllers configuration for in cluster initialization.

### PR DESCRIPTION
**Pull request by LucasStarlingdePaulaSalles**
# Description

Kind: *Story*

Section: *Controller clients*

Summary:
Created a configurations object for the controller clients, and way to get said configs from a cluster deployment

## Changelog

- fixes:
        - Renamed poorly named rest clients on controller clients.
- features:
	- ControllerConfig object developed to help initialize controllers and ease of config validation.
	- GetInClusterConfigs function implemented to fetch controller configs from current in-cluster dapp deployment
- tests:
	- unit tests for GetInClusterConfigs

### CODE REVIEW

File: pkg/controller/client/client.go [hunk 1]
```diff
@@ -2,55 +2,92 @@ package client
...
+	"github.com/inspr/inspr/pkg/auth"
 	"github.com/inspr/inspr/pkg/controller"
+	"github.com/inspr/inspr/pkg/ierrors"
 	"github.com/inspr/inspr/pkg/rest/request"
 )
 
 // Client implements communication with the Insprd
 type Client struct {
 	HTTPClient *request.Client
+	config     ControllerConfig
```
Comment (franciscobonand) [inline pkg/controller/client/client.go]: Shouldn't this field also be public?

File: pkg/controller/client/client.go [hunk 2]
```diff
@@ -2,55 +2,92 @@ package client
...
+}
+
+// GetInClusterConfigs retrieves controller configs from current dApp deployment.
+func GetInClusterConfigs() (*ControllerConfig, error) {
+	url, urlok := os.LookupEnv("INSPR_INSPRD_ADDRESS")
+	scope, scopeok := os.LookupEnv("INSPR_CONTROLLER_SCOPE")
+	token, tknok := os.LookupEnv("INSPR_CONTROLLER_TOKEN")
+	if !urlok || !scopeok || !tknok {
+		return nil, ierrors.NewError().
+			Message("authentication as controller failed. controllers requires following variables: INSPR_INSPRD_ADDRESS, INSPR_CONTROLLER_SCOPE & INSPR_CONTROLLER_TOKEN").
```
Comment (franciscobonand) [inline pkg/controller/client/client.go]: Might be a good idea to break this line into two so it's not so wide

**Review by franciscobonand — APPROVED** [review]
LGTM

File: pkg/controller/client/client.go (hunk 1 above)
Comment (LucasStarlingdePaulaSalles) [reply pkg/controller/client/client.go]: Done budy

File: pkg/controller/client/client.go (hunk 2 above)
Comment (LucasStarlingdePaulaSalles) [reply pkg/controller/client/client.go]: I tried, but it becomes an error

Comment (LucasStarlingdePaulaSalles) [reply pkg/controller/client/client.go]: never mind, figured it out

**Review by luiz-couto — APPROVED** [review]

**Pull request merged**